### PR TITLE
http: tracking object scope on the encode path

### PR DESCRIPTION
**Pull request by alyssawilk**
Tracking the active stream on the encode path, for crash logging.

Risk Level: Medium (touching the router)
Testing: new unit tests
Docs Changes: n/a
Release Notes: n/a
#7300

### CODE REVIEW

**Comment by alyssawilk** [comment]
I think this is good for review!

File: test/common/router/router_test.cc [hunk 1]
```diff
@@ -98,6 +99,8 @@ class RouterTestBase : public testing::Test {
 
     // Make the "system time" non-zero, because 0 is considered invalid by DateUtil.
     test_time_.setMonotonicTime(std::chrono::milliseconds(50));
+
+    EXPECT_CALL(callbacks_.dispatcher_, setTrackedObject(_)).Times(AnyNumber());
```
Comment (mattklein123) [inline test/common/router/router_test.cc]: Is this needed due to a strict mock? If so can you comment? If not can we remove? Same below?

File: test/common/router/router_upstream_log_test.cc [hunk 2]
```diff
@@ -91,6 +91,7 @@ class RouterUpstreamLogTest : public testing::Test {
                                    router_proto));
     router_.reset(new TestFilter(*config_));
     router_->setDecoderFilterCallbacks(callbacks_);
+    EXPECT_CALL(callbacks_.dispatcher_, setTrackedObject(_)).Times(testing::AnyNumber());
```
Comment (mattklein123) [inline test/common/router/router_upstream_log_test.cc]: Sorry can you add a similar comment here about the strict mock? Fine to do in your next change if you want.

**Pull request merged**